src/dune_analytics.py

```diff
@@ -13,8 +17,87 @@
 BASE_URL = "https://dune.xyz"
 GRAPH_URL = 'https://core-hsr.duneanalytics.com/v1/graphql'
 
+
 # --------- Constants --------- #
 
+class DuneParameterType(Enum):
+    """
+    Enum of the 4 distinct dune parameter types
+    """
+    TEXT = 'text'
+    NUMBER = 'number'
+    DATE = 'datetime'
+    LIST = 'list'
+
+
+class QueryParameter:
+    """Class whose instances are Dune Compatible Query Parameters"""
+
+    def __init__(
+            self,
+            name: str,
+            parameter_type: DuneParameterType,
+            raw_value: Any,
+            value_str: str
+    ):
+        self.key: str = name
+        self.type: DuneParameterType = parameter_type
+        self.value_str = value_str
+        self.original_value = raw_value
+
+    @classmethod
+    def text_type(cls, name: str, value: str):
+        """Constructs a Query parameter of type text"""
+        return cls(
+            name,
+            DuneParameterType.TEXT,
+            raw_value=value,
+            value_str=value,
+        )
+
+    @classmethod
+    def number_type(cls, name: str, value: int | float):
+        """Constructs a Query parameter of type number"""
+        return cls(
+            name,
+            DuneParameterType.NUMBER,
+            raw_value=value,
+            value_str=str(value),
+        )
+
+    @classmethod
+    def date_type(cls, name: str, value: datetime):
+        """Constructs a Query parameter of type date"""
+        return cls(
+            name,
+            DuneParameterType.DATE,
+            raw_value=value,
+            # This is the postgres string format of timestamptz
+            value_str=value.strftime("%Y-%m-%d %H:%M:%S"),
+        )
+
+    @classmethod
+    def list_type(cls, name: str, value: list[str]):
+        """Constructs a Query parameter of type list"""
+        return cls(
+            name,
+            DuneParameterType.LIST,
+            raw_value=value,
+            value_str="\n".join(value),
```

Comment:
Using `\n` seems surprising to me. I have no idea of dune but should that maybe be a `,`?

Reply:
<img width="372" alt="Screenshot 2022-03-04 at 12 54 44 PM" src="https://user-images.githubusercontent.com/11778116/156759083-9a6bf0e1-d2bc-4bf2-bfb7-8265b7a51529.png">
This is the specification given by Dune.